### ISSUE TITLE
[Phase 4] KD Loss 함수 구현 및 단위 테스트 ⭐

### ISSUE THREAD

**Issue by YOONPYOGitHub** [YOONPYOGitHub/knowledge-distillation#46]
## 목표
지식 증류의 핵심인 KD Loss 함수를 구현하고, 수학적으로 정확한지 검증

## 기간
Phase 4 (5~6개월차) | 예상 2주

### KD Loss 핵심 구현
- [ ] Temperature Scaling: `softmax(logits / T)`
- [ ] KL Divergence 계산: `KL(softmax(teacher/T) ∥ log_softmax(student/T))`
- [ ] T² 보정 계수 적용
- [ ] Hard Label Loss: `CrossEntropy(student_logits, labels)`
- [ ] Total Loss: `α · CE + (1-α) · T² · KL`

### 단위 테스트 — 수식 검증
- [ ] T=1일 때 soft label ≈ hard label인지 확인
- [ ] T→∞ 일 때 uniform distribution에 근접하는지 확인
- [ ] α=1.0일 때 KD loss = CE loss인지 확인
- [ ] α=0.0일 때 KD loss = T²·KL인지 확인
- [ ] 동일 Teacher/Student 입력 시 KL ≈ 0인지 확인
- [ ] Loss가 항상 ≥ 0인지 확인
- [ ] Gradient가 Student에만 흐르는지 확인 (Teacher는 frozen)

### 수도코드 → 실제 코드
```python
# README.md의 수도코드를 실제 구현
for batch in dataloader:
    with torch.no_grad():
        teacher_logits = teacher_model(input_ids).logits
    student_logits = student_model(input_ids).logits
    
    teacher_soft = F.softmax(teacher_logits / T, dim=-1)
    student_soft = F.log_softmax(student_logits / T, dim=-1)
    
    kd_loss = T * T * F.kl_div(student_soft, teacher_soft, reduction="batchmean")
    ce_loss = F.cross_entropy(student_logits.view(-1, vocab_size), labels.view(-1))
    
    total_loss = alpha * ce_loss + (1 - alpha) * kd_loss
```

## 참고
- `README.md` — 핵심 알고리즘 상세 (수도코드)
- `docs/glossary.md` — KD Loss 수식

## 완료 기준
- [ ] KD Loss 함수가 모든 단위 테스트 통과
- [ ] 수식과 코드가 1:1 대응되는지 팀 리뷰 완료
